### ISSUE TITLE
没有输出？

### ISSUE THREAD

**Issue by walsy123** [formulahendry/vscode-code-runner#167]
![20170825142333](https://user-images.githubusercontent.com/13585160/29701938-42f9ccec-89a1-11e7-927e-96c3e0e49285.png)

环境变量没问题，为何一直没有输出就结束？

**Comment by formulahendry** [comment]
Obviously, You have selected part of code snippet. Then it is running the selected code, not the whole file. You could to refer to #156 to avoid that if you do not need to run code snippet.

**Comment by walsy123** [comment]
![20170825143644](https://user-images.githubusercontent.com/13585160/29702221-dd719132-89a2-11e7-9933-a66746f079b4.png)
运行整个文件，也是一样没有输出。
如果输出到终端则没问题。为什么呢？

**Comment by formulahendry** [comment]
Do you have changed any other settings?

**Comment by walsy123** [comment]
没有修改任何 code runner 设置。卸载重新安装还是没有输出。 - -

**Comment by formulahendry** [comment]
Did you try running a simple 'Hello World' PHP program? Or try running BAT, JS or other file?

**Comment by walsy123** [comment]
![1](https://user-images.githubusercontent.com/13585160/29704535-d2dc6d00-89ac-11e7-8882-e81dc4344135.png)
![2](https://user-images.githubusercontent.com/13585160/29704536-d2e1ff18-89ac-11e7-9af9-09a81c97a691.png)
php 和 js 简单的打印，还是这样。

**Comment by formulahendry** [comment]
How about trying below setting (`File`->`Preference`->`Settings`) to run code in Integrated Terminal:
```json
{
    "code-runner.runInTerminal": true
}
```

**Comment by formulahendry** [comment]
BTW, what's your VS Code version?

**Comment by walsy123** [comment]
![3](https://user-images.githubusercontent.com/13585160/29705576-4690224c-89b1-11e7-8b2d-c15fdf6715ad.png)
![4](https://user-images.githubusercontent.com/13585160/29705575-465ce09e-89b1-11e7-8749-a11a27d42e50.png)
![5](https://user-images.githubusercontent.com/13585160/29705577-46abf9f4-89b1-11e7-8424-d543a518ee8f.png)
输出到终端是正常的。

**Comment by formulahendry** [comment]
Could you go to `Help` > `Toggle Developer Tools` to see if there is any error in console?

**Comment by walsy123** [comment]
![6](https://user-images.githubusercontent.com/13585160/29706287-ef86b67a-89b3-11e7-8816-0edf570a72a1.png)
跟这个有关系吗
输出终端始终正常。

**Comment by formulahendry** [comment]
Are there any other errors?

**Comment by walsy123** [comment]
run code 的时候没有 任何错误提示。

**Comment by WongSeon** [comment]
@formulahendry
嘿嘿 我用这个编写Java的, 之前输出的时候总会出现乱码.

但是, 当我在个人自定义设置里添加如下设置的时候(依旧是"false"):
"code-runner.runInTerminal": false

乱码的问题解决了, 你可以试下啊, 如果其他语言也行, 可以在指导说明里教下其他不会的人, 而不是去使用系统的命令行, 不够简洁.

谢谢

**Comment by formulahendry** [comment]
@WongSeon , 这个不太科学啊。那你现在再把`"code-runner.runInTerminal": false`删掉呢？又是乱码了？

**Comment by aojie654** [comment]
@formulahendry 我倒是发现了一个问题，win7 7601,vs code 1.18.1，code runner 0.8.3，设置code-runner.runInTerminal": false 是在输出窗口中进行输出，这时候python输出中文乱码，c和java正常，但是如果是code-runner.runInTerminal": true，则在终端中输出，python正常，c和java乱码。
